src/components/UI/PageMetadata.tsx

```diff
@@ -14,10 +16,15 @@ export const PageMetadata: FC<Props> = ({
   description,
   image = HOMEPAGE_HERO_MOBILE_URL
 }) => {
+  const { asPath } = useRouter();
   const fullTitle = `${title} | ${HEAD_TITLE}`;
+
+  const canonicalUrl = getFullUrl(asPath);
   return (
     <Head>
       <title>{fullTitle}</title>
+      <link rel='canonical' href={canonicalUrl} />
```

Comment:
FYI I'm also adding the correct canonical urls for nested pages since we were always pointing to the root page.